spec/lib/appsignal_spec.rb

```diff
@@ -1297,7 +1297,7 @@ def on_start
             transaction._sample
             expect(transaction).to have_namespace("my_namespace")
             expect(transaction).to have_action("my_action")
-            expect(transaction).to have_error("ExampleException", "error message")
+            expect(transaction.errors).to eq([error])
```

Comment:
The error is now being set as one of the additional errors in the `@errors` array (so `have_error` cannot match it -- it's not set on the transaction yet)

As currently implemented, the error won't be set into the transaction until the transaction is completed -- at which point `.to_h` will return `nil`, so we can't check it then, either.

(It would be possible to change the implementation to set it immediately if `@has_error` is false? I'll look into that)

Reply:
> (It would be possible to change the implementation to set it immediately if `@has_error` is false? I'll look into that)

I think we set the error on the transaction in the extension if none has been set yet. And if it has, add it to the `@errors` list.

Reply:
Yes, we could do that. This has an issue in that, if something else calls `set_error` (and not `add_error`) then it will overwrite that error.

I think to fix this we should make `set_error` be an alias of `add_error`. I'll give it a try.

Reply:
Ah, that's an annoying side effect. But it's kind of what people expect to happen now anyway. Our plan of deprecating `Transaction#set_error`, and the public `Appsignal.set_error`, seems good and solves this problem in the future.

Reply:
Does [this diff](https://github.com/appsignal/appsignal-ruby/compare/multiple-errors...multiple-errors-aliased) make sense? It's not so much deprecating it as just changing its behaviour into what people already expect.

Let me know and I'll add it to this PR.

Reply:
@unflxw looks good. You can add it to the PR. I might refactor out the `_set_error` private method testing.

Reply:
Added -- feel free to look into that refactor.